include/nbl/builtin/hlsl/cpp_compat/vector.h

```diff
@@ -15,6 +15,18 @@ using vector = glm::vec<N, T>;
 using float4 = vector<float, 4>;
 using float3 = vector<float, 3>;
 using float2 = vector<float, 2>;
+using float1 = vector<float, 1>;
+
+using int4 = vector<int32_t, 4>;
+using int3 = vector<int32_t, 3>;
+using int2 = vector<int32_t, 2>;
+using int1 = vector<int32_t, 1>;
+
+using uint4 = vector<uint32_t, 4>;
+using uint3 = vector<uint32_t, 3>;
+using uint2 = vector<uint32_t, 2>;
+using uint1 = vector<uint32_t, 1>;
```

Comment:
having `T1` as `vector<T,1>` instead of `T` is a mistake in my opinion.

Reply:
also I'd prefer `int32_tN` instead of `intN` and friends, since they've been introduced in HLSL2021

Reply:
ok reoslved, did it myself check the PR I opened #559